### PR TITLE
Refactoring some problems for better readability, modularity and main…

### PR DESCRIPTION
**Pull request by EMAYAN08**
Refactoring some problems for better readability, modularity and maintainability

Refactoring changes also passed the respective test cases

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -193,5 +193,11 @@
             <version>[2.1.1,)</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.13.1</version>
```
Comment (javadev) [inline pom.xml:199]: Why junit 4, we use junit 5

File: src/main/java/g1401_1500/s1496_path_crossing/Solution.java [hunk 2]
```diff
@@ -1,7 +1,5 @@
 package g1401_1500.s1496_path_crossing;
 
-// #Easy #String #Hash_Table #2022_03_23_Time_1_ms_(97.41%)_Space_42.9_MB_(31.41%)
```
Comment (javadev) [inline src/main/java/g1401_1500/s1496_path_crossing/Solution.java:3]: Why tags were removed?

File: src/main/java/g2101_2200/s2141_maximum_running_time_of_n_computers/Solution.java [hunk 3]
```diff
@@ -4,31 +4,34 @@
 // #2022_06_05_Time_24_ms_(70.39%)_Space_83.1_MB_(5.83%)
```
Comment (javadev) [inline src/main/java/g2101_2200/s2141_maximum_running_time_of_n_computers/Solution.java:4]: We need to update tags after improve solution

File: src/test/java/g1901_2000/s1906_minimum_absolute_difference_queries/SolutionTest.java [hunk 4]
```diff
@@ -1,28 +1,49 @@
...
-class SolutionTest {
     @Test
-    void minDifference() {
-        assertThat(
-                new Solution()
-                        .minDifference(
-                                new int[] {1, 3, 4, 8},
-                                new int[][] {{0, 1}, {1, 2}, {2, 3}, {0, 3}}),
-                equalTo(new int[] {2, 1, 4, 1}));
+    public void testSingleElementInput() {
```
Comment (javadev) [inline src/test/java/g1901_2000/s1906_minimum_absolute_difference_queries/SolutionTest.java:9]: We use other name convention for tests

**Comment by javadev** [comment]
Hello @EMAYAN08 

Thank you for your pull request.

We need to update tags after improve solution.

Please use generamd tool

**Comment by javadev** [comment]
Please format sources with help of gradle spotless plugin

**Comment by EMAYAN08** [comment]
> Please format sources with help of gradle spotless plugin

Okay, I will do the needed changes and update the pull request

**Comment by EMAYAN08** [comment]
Thank you for your time @javadev

File: src/main/java/g1401_1500/s1496_path_crossing/Solution.java [hunk 5]
```diff
@@ -41,15 +39,24 @@ public boolean isPathCrossing(String path) {
...
-        int y;
+        private final int x;
+        private final int y;
 
         public Coord(int x, int y) {
             this.x = x;
             this.y = y;
         }
 
+        public int getX() {
```
Comment (javadev) [inline src/main/java/g1401_1500/s1496_path_crossing/Solution.java:52]: This class under sonarcloud control, no changes needed

File: src/main/java/g1901_2000/s1906_minimum_absolute_difference_queries/Solution.java [hunk 6]
```diff
@@ -1,106 +1,32 @@
 package g1901_2000.s1906_minimum_absolute_difference_queries;
 
 // #Medium #Array #Hash_Table #2022_05_11_Time_176_ms_(71.61%)_Space_82.6_MB_(95.48%)
-
 import java.util.Arrays;
 import java.util.BitSet;
+import java.util.Scanner;
 
 public class Solution {
-    private static class SegmentTree {
```
Comment (javadev) [inline src/main/java/g1901_2000/s1906_minimum_absolute_difference_queries/Solution.java:9]: We use embedded classes for our html version https://leetcode-in-java.github.io

**Comment by javadev** [comment]
There are too many changes. Please create separate pull request for each task + spotless + tags

**Comment by EMAYAN08** [comment]
> 

Thank you, I'll look into it @javadev

**Pull request closed without merging**